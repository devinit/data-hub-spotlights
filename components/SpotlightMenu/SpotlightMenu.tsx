import React, { FunctionComponent } from 'react';

const SpotlightMenu: FunctionComponent = ({ children }) => {
  return <div className="spotlight__menu">{children}</div>;
};

export { SpotlightMenu };
