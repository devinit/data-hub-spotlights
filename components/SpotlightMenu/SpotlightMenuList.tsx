import React, { Children, cloneElement, isValidElement, FunctionComponent } from 'react';

interface SpotlightMenuListProps {
  active?: boolean;
  classNames?: string;
  onViewClick?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, title?: string) => void;
}

const SpotlightMenuList: FunctionComponent<SpotlightMenuListProps> = ({
  active,
  children,
  classNames,
  onViewClick
}) => {
  return (
    <ul className={classNames}>
      {Children.map(children, child => isValidElement(child) && cloneElement(child, { onView: onViewClick }))}
      <style jsx>{`
        display: ${active ? 'block !important' : 'none'};
        .countries-menu-list__content {
          max-height: 40vh;
        }
      `}</style>
    </ul>
  );
};

export { SpotlightMenuList };
