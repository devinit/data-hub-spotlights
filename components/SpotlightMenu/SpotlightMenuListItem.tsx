import classNames from 'classnames';
import React, { Children, FunctionComponent, useState, cloneElement, isValidElement } from 'react';

interface ComponentProps {
  title?: string;
  depth?: number;
  onView?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, title?: string) => void;
}

const SpotlightMenuListItem: FunctionComponent<ComponentProps> = ({ title, children, onView: onViewProp, depth }) => {
  const [active, setActive] = useState(false);
  const toggleActive = (): void => {
    setActive(!active);
  };
  const onView = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>): void => {
    event.stopPropagation();
    if (onViewProp) {
      onViewProp(event, title);
    }
  };

  return (
    <li
      className={classNames({
        'countries-menu-list--has-children': children,
        'countries-menu-list__countries': !children,
        'js-profile-country-item': !children
      })}
    >
      <a
        className={classNames('countries-menu-list__item js-menu-item js-search-item', {
          active,
          'countries-menu-list__item--open': active && children,
          'countries-menu-list__item--parent-first': depth === 1,
          'countries-menu-list__item--parent-second': depth === 2,
          'countries-menu-list__item--parent-third': depth === 3,
          'countries-menu-list__item--parent-fourth': depth === 4,
          'countries-menu-list__item--parent-fifth': depth === 5,
          'countries-menu-list__item--parent-sixth': depth === 6
        })}
        onClick={toggleActive}
      >
        {title}
        <style jsx>{`
          :before {
            display: ${children ? 'inline-block' : 'none'};
          }
        `}</style>
      </a>
      <a
        className="countries-menu__profile countries-menu__link js-profile-item"
        onClick={onView}
        title={`View ${title}`}
      >
        View
      </a>
      {Children.map(children, child => isValidElement(child) && cloneElement(child, { active }))}
    </li>
  );
};

SpotlightMenuListItem.defaultProps = { depth: 1 };

export { SpotlightMenuListItem };
