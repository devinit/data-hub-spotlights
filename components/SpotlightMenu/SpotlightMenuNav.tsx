import classNames from 'classnames';
import React, { Children, FunctionComponent, isValidElement, cloneElement } from 'react';

interface ComponentProps {
  active?: boolean;
  onClick?: () => void;
  caption: string;
  onShowAll?: () => void;
}

const SpotlightMenuNav: FunctionComponent<ComponentProps> = ({ active, caption, children, onClick, onShowAll }) => {
  return (
    <nav className={classNames('countries-menu-list animated', { inactive: !active })}>
      <a
        className="countries-menu-list__item countries-menu-list__parent countries-menu-list__item--open js-countries-menu-trigger"
        onClick={onClick}
      >
        {caption}
      </a>
      <a
        className={classNames('countries-menu__profile countries-menu__link js-profile-item', { hide: !onShowAll })}
        aria-hidden="true"
        title={`View ${caption}`}
        onClick={onShowAll}
        data-testid="spotlight-nav-view"
      >
        View
      </a>
      <style jsx>{`
        cursor: pointer;
      `}</style>
      {Children.map(children, (child) => isValidElement(child) && cloneElement(child, { active }))}
    </nav>
  );
};

export default SpotlightMenuNav;
