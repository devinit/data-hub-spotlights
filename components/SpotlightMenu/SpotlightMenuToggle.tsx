import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

interface ComponentProps {
  show?: boolean;
  onClick?: () => void;
  caption: string;
}

const SpotlightMenuToggle: FunctionComponent<ComponentProps> = ({ show, caption, onClick }) => {
  return (
    <nav className={classNames('countries-menu-list js-countries-menu-trigger', { inactive: !show })}>
      <a className="countries-menu-list__item countries-menu-list__parent" onClick={onClick}>
        <span>{caption}</span>
        <style jsx>{`
          cursor: pointer;
          text-transform: capitalize;
        `}</style>
      </a>
    </nav>
  );
};

SpotlightMenuToggle.defaultProps = { show: false };

export { SpotlightMenuToggle };
