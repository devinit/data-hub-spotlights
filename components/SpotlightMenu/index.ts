export * from './SpotlightMenu';
export * from './SpotlightMenuList';
export * from './SpotlightMenuListItem';
export * from './SpotlightMenuNav';
export * from './SpotlightMenuToggle';
