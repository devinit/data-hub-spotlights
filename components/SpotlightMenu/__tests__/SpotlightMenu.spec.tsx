/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import { SpotlightMenu, SpotlightMenuItem } from '../SpotlightMenu';

describe('SpotlightMenu', () => {
  const sidebarItems: SpotlightMenuItem[] = [
    {
      title: 'Level 1',
      children: [
        {
          title: 'Level 1.1'
        },
        {
          title: 'Level 1.2'
        },
        {
          title: 'Level 1.3'
        }
      ]
    },
    {
      title: 'Level 2'
    },
    {
      title: 'Level 3',
      children: [
        {
          title: 'Level 3.1'
        },
        {
          title: 'Level 3.2',
          url: 'https://google.com'
        }
      ]
    }
  ];

  test('renders correctly', () => {
    const renderer = TestRenderer.create(
      <SpotlightMenu items={ sidebarItems }/>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
