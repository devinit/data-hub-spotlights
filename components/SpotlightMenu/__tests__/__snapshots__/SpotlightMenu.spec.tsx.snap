// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightMenu renders children correctly 1`] = `
<div
  className="spotlight__menu"
>
  <div>
    Testing
  </div>
</div>
`;

exports[`SpotlightMenu renders correctly 1`] = `
<div
  className="spotlight__menu"
/>
`;
